luaui/Widgets/Tests/cmd_blueprint/test_cmd_blueprint_filter.lua

```diff
@@ -0,0 +1,109 @@
+local widgetName = "Blueprint"
+
+function skip()
+	return Spring.GetGameFrame() <= 0
+end
+
+function setup()
+	assert(widgetHandler.knownWidgets[widgetName] ~= nil)
+
+	Test.clearMap()
+
+	initialWidgetActive = widgetHandler.knownWidgets[widgetName].active
+	if initialWidgetActive then
+		widgetHandler:DisableWidget(widgetName)
+	end
+	widgetHandler:EnableWidget(widgetName, true)
+
+	initialCameraState = Spring.GetCameraState()
+
+	Spring.SetCameraState({
+		mode = 5,
+	})
+end
+
+function cleanup()
+	Test.clearMap()
+
+	widgetHandler:DisableWidget(widgetName)
+	if initialWidgetActive then
+		widgetHandler:EnableWidget(widgetName, false)
+	end
```

Comment:
You can remove lines 28-31 here, the widget restoring will be done automatically by the test runner when you use prepareWidget(widgetName) at setup().